EHospital.Authorization.Data/Data/UsersDataContext.cs

```diff
@@ -0,0 +1,268 @@
+﻿namespace EHospital.Authorization.Data
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using EHospital.Authorization.Model;
+    using EHospital.Authorization.WebAPI;
+    using Microsoft.EntityFrameworkCore;
+
+    public class UsersDataContext : DbContext, IDataProvider
+    {
+        public UsersDataContext(DbContextOptions<UsersDataContext> options) : base(options) { }
+
+        public DbSet<UsersData> UsersData { get; set; }
+
+        public DbSet<Logins> Logins { get; set; }
+
+        public DbSet<Roles> Roles { get; set; }
+
+        public DbSet<Secrets> Secrets { get; set; }
+
+        public DbSet<Sessions> Sessions { get; set; }
+
+        public string Token { get; set; }
+
+        public async Task AddRoles(Roles roles)
+        {
+            {
+                this.Roles.Add(roles);
+               await this.SaveChangesAsync();
+            }
+        }
+
+        public async Task AddLogin(Logins login)
```

Comment:
All 'Add' methods is calling from one place in Registration controller. So if they all are needed for user creating you can remove from them await this.SaveChangesAsync(); and call saveChanges once from registrationController. In this way, you will make only one save call to the db, not 4 or 5. And It can be covered with transaction.